### ISSUE TITLE
Add option for disable chat in public shares

### ISSUE THREAD

**Issue by Ivansss** [nextcloud/spreed#2284]
Related with #2280 
Apart from the global options in #2283, it would be nice to have an option to disable chat in public shares for a specific file.
This option will be available only for the file owner and enabled by default.
If the file owner disables this option, only internal users/shares will be able to use the chat attached to this file and the chat will not appear in public shares.

<img width="509" alt="Screenshot 2019-10-08 at 16 35 23" src="https://user-images.githubusercontent.com/4638605/66406557-1b3c4880-e9ec-11e9-881b-6b3dc0b5d2ed.png">

@nickvergessen 
@nextcloud/designers ideas?

**Comment by Ivansss** [comment]
@jancborchardt we thought about adding it to the "share link" menu too :)
But the problem is that the chat is not share related but file related.
In other words, there is just one chat per file, not one chat per share.
That way, all the sharees use the same chat for collaboration.
So this option should be outside of the "share link" menu.

**Comment by danxuliu** [comment]
It could still be added to the _Sharing_ tab outside all shares, but I think that it will be strange.

But there was an additional reason to add it to the Chat tab, which is not touching the server code for this. As far as I know right now apps can not extend the _Sharing_ tab, so the option would need to be added in the server itself, and I would like to avoid as much as possible to add app specific code to the server (yes, the _Password protect by Talk_ option is hardcoded in the server, but in that case there was no other way around it ;-) ).

Also, another phrasing (which I think is better) that we came up for the option was _Enable public conversations_ (instead of _Enable chat in public shares_).

**Comment by Ivansss** [comment]
> Also, another phrasing (which I think is better) that we came up for the option was Enable public conversations (instead of Enable chat in public shares).

I know that we came up with "Enable public conversations" 😄 , but then I thought that this wording might be hard to relate with the chat shown in public shares.

Another idea could be "Enable chat in shared links"? So we use a similar term as in Files.
<img width="240" alt="Screenshot 2019-10-09 at 09 27 26" src="https://user-images.githubusercontent.com/4638605/66460446-3b651980-ea77-11e9-8dc4-8f62c856a728.png">

**Comment by nickvergessen** [comment]
@jancborchardt 
> With the mockup by juliushaertl on hiding the advanced settings, we can put it in there

I think it is somewhat weird to have it in there, if it manages all shares for the same file. And I can almost guarantee that there will be a request to manage them individually and that will be a pain to oversight and manage in the backends...

**Comment by jancborchardt** [comment]
> I think it is somewhat weird to have it in there, if it manages all shares for the same file.

No, the menu exists for every share link individually, including mail shares (and we could even have it in the shares with people on the same instance).

So you could e.g. send a link to one group which makes the chat visible and actionable, and to another who only sees the file itself.


@Ivansss And yes, it should simply be called "Enable chat". Links are not "public" per se, so we should not call it that or people think it will be made public on some list. ;)

**Comment by nickvergessen** [comment]
The share table has no support for this, so it will become a mess. Exactly why I think we should stick to the global setting for now. The details for this minor change is just too much effort at the moment

**Comment by jancborchardt** [comment]
@ma12-co @jenniferpiperek do you understand this, and/or do you have a proposal here? :)

**Comment by ma12-co** [comment]
Well from what I understand it all boils down on back-end support for this right? 

As per the busy popover, I think we could also explore a horizontal navigation flow similar to the color picker one, something like:

![IMG_20191010_082613](https://user-images.githubusercontent.com/26852655/66544349-2dc39880-eb38-11e9-9ee6-fa041fffa491.jpg)

**Comment by tigernero79** [comment]
I saw that talk 7.0.1 update was released which should include the ability to disable chat in the URL of the file sharing but I can't find how to do it is the option?

**Comment by nickvergessen** [comment]
> I saw that talk 7.0.1 update was released which should include the ability to disable chat in the URL of the file sharing but I can't find how to do it is the option?

It's in the admin settings, visit `index.php/settings/admin/talk` on your instance as an admin

**Comment by tigernero79** [comment]
thanks, but option came out only with such 7.0.2
